src/main/java/at/hannibal2/skyhanni/features/chat/StashCompact.kt

```diff
@@ -0,0 +1,120 @@
+     */
+    private val pickupStashPattern by patternGroup.pattern(
+        "pickup.stash",
+        "§f *§3§l>>> §3§lCLICK HERE§b to pick (?:them|it) up! §3§l<<<.*",
+    )
+
+    /**
+     * REGEX-TEST: §eOne or more items didn't fit in your inventory and were added to your item stash! §6Click here to pick them up!
+     * REGEX-TEST: §eOne or more materials didn't fit in your inventory and were added to your material stash! §6Click here to pick them up!
+     */
+    @Suppress("MaxLineLength")
+    private val genericAddedToStashPattern by patternGroup.pattern(
+        "generic",
+        "§eOne or more (?:item|material)s? didn't fit in your inventory and were added to your (?:item|material) stash! §6Click here §eto pick them up!",
+    )
+    //</editor-fold>
+
+    private val config get() = SkyHanniMod.feature.chat.filterType.stashMessages
+
+    private var lastMaterialCount = 0
+    private var lastDifferingMaterialsCount = 0
+    private var lastType = ""
+
+    private var lastSentMaterialCount = 0
+    private var lastSentType = ""
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+
+        genericAddedToStashPattern.matchMatcher(event.message) {
+            event.blockedReason = "stash_compact"
+        }
+
+        materialCountPattern.matchMatcher(event.message) {
+            groupOrNull("count")?.replace(",", "")?.toIntOrNull()?.let { count ->
+                lastMaterialCount = count
+            }
+            event.blockedReason = "stash_compact"
+        }
+
+        differingMaterialsCountPattern.matchMatcher(event.message) {
+            groupOrNull("count")?.replace(",", "")?.toIntOrNull()?.let { count ->
+                lastDifferingMaterialsCount = count
+            }
+            groupOrNull("type")?.let { type ->
+                lastType = type
+            }
+            event.blockedReason = "stash_compact"
+        }
+
+        if (pickupStashPattern.matches(event.message)) {
+            event.blockedReason = "stash_compact"
+            if (lastMaterialCount <= config.hideLowWarningsThreshold) return
+            if (config.hideDuplicateCounts && lastMaterialCount == lastSentMaterialCount && lastType == lastSentType) return
+
+            sendCompactedStashMessage()
+        }
+    }
+
+    private fun sendCompactedStashMessage() {
+        ChatUtils.clickableChat(
+            "§7You have §3${lastMaterialCount} §7${StringUtils.pluralize(lastMaterialCount, lastType)} in stash, " +
+                "§8totalling $lastDifferingMaterialsCount ${StringUtils.pluralize(lastDifferingMaterialsCount, "type")}§7. " +
+                "§3Click to ${if(config.useViewStash) "view stash" else "pickup stash"}§7.",
+            onClick = {
+                if (config.useViewStash) HypixelCommands.viewStash(lastType)
+                else HypixelCommands.pickupStash()
+            },
+            prefix = false,
```

Comment:
Users will have a hard time finding this feature when they want to disable it. It's default enabled and has no prefix. Maybe we should still show the prefix?